src/train_softmax.py

```diff
@@ -205,7 +205,16 @@ def main(args):
 
             if pretrained_model:
                 print('Restoring pretrained model: %s' % pretrained_model)
-                saver.restore(sess, pretrained_model)
+                if args.checkpoint_exclude_scopes:
+                    exclusions = [args.checkpoint_exclude_scopes.split(',')]
+                else:
+                    exclusions = []
+                # load from pre-trained model
+                tf.contrib.framework.assign_from_checkpoint_fn(
+                    pretrained_model,
+                    [var for var in tf.trainable_variables()
+                        if all(not var.op.name.startswith(exclusion) for exclusion in exclusions)],
+                    ignore_missing_vars=args.ignore_missing_vars)(sess)
```

Comment:
Hi, I'm confused of this piece of code, why follows(sess) after invoke tf.contrib.framework.assign_from_checkpoint_fn?